### PR TITLE
enable training mask2former and maskformer for transformers trainer

### PR DESCRIPTION
**Pull request by SangbumChoi**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [X] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

Originally the output of loss_dict was like as follows
```
{'loss_mask': tensor([2.1010], device='cuda:1', grad_fn=<MulBackward0>), 'loss_dice': tensor([2.0065], device='cuda:1', grad_fn=<MulBackward0>), 'loss_cross_entropy': tensor(8.3828, device='cuda:1', grad_fn=<MulBackward0>), 'loss_mask_0': tensor([1.5738], device='cuda:1', grad_fn=<MulBackward0>), 'loss_dice_0': tensor([3.0623], device='cuda:1', grad_fn=<MulBackward0>), 'loss_cross_entropy_0': tensor(8.3828, device='cuda:1', grad_fn=<MulBackwa
```
which returns total loss as 1 dimension shape of tensor like tensor([]).
However, transformers trainer https://github.com/huggingface/transformers/blob/3cefac1d974db5e2825a0cb2b842883a628be7a0/src/transformers/trainer.py#L1772
requires float tensor type. From the [example](https://github.com/NielsRogge/Transformers-Tutorials/blob/master/MaskFormer/Fine-tuning/Fine_tune_MaskFormer_on_an_instance_segmentation_dataset_(ADE20k_full).ipynb) you may notice that output loss was returning one dimensional shape.

The reason was         
num_masks_pt = torch.as_tensor([num_masks], dtype=torch.float, device=device)
returns one dimensional shape which can be also single integer tensor.

This will not make any difference of training/inference logic but enables to train with transformers trainer.

@muellerzr and @pacman100

### CODE REVIEW

**Review by NielsRogge — APPROVED** [review]
Thanks so much! Weird that the loss wasn't a scalar.

**Comment by SangbumChoi** [comment]
FYI, original mask2former/maskformer is also not scalar. However, this actually doesn't matter at the `loss.backward()`

**Review by ArthurZucker — APPROVED** [review]
Thanks for the fix, testing was updated accordingly so LGTM
